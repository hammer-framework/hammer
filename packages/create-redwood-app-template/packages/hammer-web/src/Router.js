import React from 'react'
import {
  BrowserRouter,
  Route as RealRoute,
  Switch,
  Redirect,
  Link,
} from 'react-router-dom'

import { useAuth } from './HammerProvider'

// Routes that are passed in as children to `Private` get their `authRequired`
// prop to `true`.
export const Private = ({ children }) => {
  return React.Children.map(children, (child) => {
    return React.cloneElement(child, { authRequired: true })
  })
}

// `PrivateRoute` will wait to determine if a user is authenticated
// before routing them to `redirectTo`, the default `login URL` or
// rendering the component
// passed to the route.
//
// <Router>
//   <Route path="/" />
//   <Private>
//     <Route path="/invoices" component={Secret} />
//     <Route path="/invoice/:id" component={Secret} />
//   </Private>
// </Router>
export const PrivateRoute = ({ path, redirectTo, ...rest }) => {
  const { loading, isAuthenticated, loginWithRedirect } = useAuth()

  if (loading) {
    // TODO: Replace this with a promise when suspense is around.
    return null
  }

  if (!isAuthenticated) {
    if (redirectTo) {
      return <Redirect to={redirectTo} />
    }

    loginWithRedirect({
      appState: { targetUrl: path },
    })
    return null
  }

  return <RealRoute path={path} {...rest} />
}

export const Route = ({ authRequired, ...rest }) => {
  return authRequired ? <PrivateRoute {...rest} /> : <RealRoute {...rest} />
}

export { BrowserRouter, Switch, Redirect, Link }
