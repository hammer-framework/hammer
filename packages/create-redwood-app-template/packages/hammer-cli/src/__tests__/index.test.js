import React from 'react';
import { render } from 'ink-testing-library';

import Router from '../index';

describe('hammer-cli', () => {
  const commands = [
    {
      default: () => 'I am the a',
      commandProps: {
        name: 'command_a',
        alias: 'a',
        description: 'the cat was frightened by the loud sound',
      },
    },
    {
      default: () => 'I am the b',
      commandProps: {
        name: 'command_b',
        alias: 'b',
        description:
          "the sound of the tyres on the road roared above that of it's engine.",
      },
    },
  ];

  const renderComponent = props =>
    render(<Router commands={commands} args={undefined} {...props} />);

  it('the default menu is shown when no arguments are passed', () => {
    const { lastFrame } = renderComponent();
    expect(lastFrame()).toMatchSnapshot();
  });

  it('routes to the correct command when the name is matched ', () => {
    const { lastFrame } = renderComponent({
      args: { _: ['command_a'] },
    });
    expect(lastFrame()).toMatch(/I am the a/g);
  });
});
