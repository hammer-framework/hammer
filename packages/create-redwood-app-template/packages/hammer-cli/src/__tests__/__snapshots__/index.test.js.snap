// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Router the default menu is shown when no arguments are passed 1`] = `
"[38;2;255;217;0m[1mâš’ Hammer[22m - Build something. (https://example.org)[39m [38;2;153;153;153m| v0.0.0-alpha.0[39m

[4m[1mCommands[22m[24m

 [1mcommand_a[22m [39mthe cat was frightened by the loud sound[22m
 [1mcommand_b[22m [39mthe sound of the tyres on the road roared above that of it's engine.[22m"
`;
